src/Features/Core/Portable/EmbeddedLanguages/RegularExpressions/LanguageServices/RegexSyntaxClassifier.cs

```diff
@@ -327,7 +327,10 @@ public void Visit(RegexPosixPropertyNode node)
             }
 
             public void Visit(RegexAlternationNode node)
-                => AddClassification(node.BarToken, ClassificationTypeNames.RegexAlternation);
+            {
+                for (var i = 1; i < node.SequenceList.NodesAndTokens.Length; i += 2)
+                    AddClassification(node.SequenceList.NodesAndTokens[i].Token, ClassificationTypeNames.RegexAlternation);
```

Comment:
What would happen for cases like `a|b||c|d|e`? Would it reach this code path?

Reply:
> What would happen for cases like a|b||c|d|e? Would it reach this code path 

Yes it would reach this (and we have tests).  This is fine.  You just get an empty sequence node between `b||c`.  We have such a concept in the regex parser:

```c#
    /// <summary>
    /// Represents a possibly-empty sequence of regex expressions.  For example, the regex ""
    /// will produce an empty RegexSequence nodes, and "a|" will produce an alternation with an
    /// empty sequence on the right side.  Having a node represent the empty sequence is actually
    /// appropriate as these are legal regexes and the empty sequence represents 'a pattern
    /// that will match any position'.  Not having a node for this would actually end up 
    /// complicating things in terms of dealing with nulls in the tree.
    /// 
    /// This does not deviate from Roslyn principles.  While nodes for empty text are rare, they
    /// are allowed (for example, OmittedTypeArgument in C#).
    /// </summary>
    internal sealed class RegexSequenceNode : RegexExpressionNode
```